File: RankLib-file-gen.R
make.file <- function(infile, outfile) {
	load(infile)
	f <- file("ranklibfile")

	genes <- dim(essentiality)[[1]]
	samples <- dim(essentiality)[[2]]
	expression.genes <- 10 #dim(expression)[[1]]
	copy.number.genes <- 10 #dim(copy.number)[[1]]
	#cat("test \n", file = f)
	for(i in 1:10) {
		#gene.name <- rownames(essentiality)[[i]]
		for(j in 1:samples) {
			#cat("gene is ", gene.name, "j = ", j, "\n")
			line <- ""
			paste(line, as.character(essentiality[i,j]), sep="")
			paste(line, "test", sep="")
			cat(line, "\n")
			#feature1 <- expression[gene.name, j]
			#feature2 <- copy.number[gene.name, j]
			for(k in 1:copy.number.genes) {
				paste(line, " ", sep="")
				paste(line, as.character(k), sep="")
				#cat(line, "\n")
				paste(line, ":", sep="")
				paste(line, as.character(copy.number[k, j]), sep="")
			}
			for(k in 1:expression.genes) {
				paste(line, " ", sep="")
				paste(line, as.character(k+copy.number.genes), sep="")
				paste(line, ":", sep="")
				paste(line, as.character(expression[k, j]), sep="")
			}
			cat(line, "\n")
			writeLines(line, f)

		}
	}
	close(f)
}




	

File: RankLib-file-gen.py
def main(): 
	f = open("data/Achilles_v2.9_training.gct")
	essentiality = readfile(f)
	print(essentiality)


def readfile(f):
	f.readline()
	dimensions = f.readline()
	dimensions = dimensions.split("\t")
	genes = 10 #int(dimensions[0])
	samples = 10 #int(dimensions[1])
	f.readline()
	mat = []
	for i in range(0, genes):
		mat.append([])
		line = f.readline().split("\t")
		for j in range(0, samples):
			mat[i].append(float(line[j+2]))
	return mat


			
if __name__ == "__main__":
    main()
	
	
